src/main/java/org/cactoos/text/TextEnvelope.java

```diff
@@ -23,71 +23,46 @@
  */
 package org.cactoos.text;
 
-import java.io.IOException;
-import org.cactoos.Scalar;
 import org.cactoos.Text;
-import org.cactoos.scalar.And;
-import org.cactoos.scalar.IoChecked;
-import org.cactoos.scalar.Or;
-import org.cactoos.scalar.Unchecked;
 
 /**
- * Text envelope that provides {@link #equals(Object)} and {@link #hashCode()}
- * methods.
+ * {@link Text} envelope.
+ *
  * @since 0.32
  * @checkstyle AbstractClassNameCheck (500 lines)
  */
 public abstract class TextEnvelope implements Text {
 
     /**
-     * String value of the envelope.
+     * Wrapped Text.
      */
-    private final IoChecked<String> origin;
+    private final Text origin;
 
     /**
      * Ctor.
      * @param text Text representing the text value.
      */
     public TextEnvelope(final Text text) {
-        this(new IoChecked<>(text::asString));
-    }
-
-    /**
-     * Ctor.
-     * @param scalar Scalar representing the text value.
-     */
-    public TextEnvelope(final Scalar<String> scalar) {
-        this.origin = new IoChecked<>(scalar);
+        this.origin = text;
     }
 
     @Override
-    public final String asString() throws IOException {
-        return this.origin.value();
+    public final String asString() throws Exception {
+        return this.origin.asString();
     }
 
     @Override
     public final String toString() {
-        return new UncheckedText(this).asString();
+        return this.origin.toString();
     }
 
     @Override
-    public final int hashCode() {
-        return new Unchecked<>(this.origin).value().hashCode();
+    public final boolean equals(final Object obj) {
+        return this.origin.equals(obj);
```

Comment:
@victornoel Wouldn't this definition of `equals` cause problems? The operation seems to loose symmetric properties. 
For example
`new TextEnvelope(x).equals(x) == true`, but  not necessarily `x.equals(new TextEnvelope(x))`

Reply:
@andreoss this is a known limitation of `equals` actually when an interface has many different implementations. This subject has been discussed many times in the EO community, i.e., what does method equals mean for objects (you can search for "equals" in cactoos issues for example).

My take on it is that if you start having multiple implementations of the same interface, then you can't just expect equals to rely on some internal state, and thus, when you implement it, you have to rely on the exposed behaviour of the interface.

In the case of `Text`, it means that all implementations of `Text` should rely on `Text.asString` to implement `equals`, so in that case, the symmetry will be conserved. It is the same for `MapOf`, `IterableOf` and the other collection-related classes for example. Also sometimes it's better not to implement `equals` at all (e.g., `Iterator`).

Reply:
@andreoss see for example https://github.com/yegor256/cactoos/issues/1156#issuecomment-511189091 where I made the same kind of comment. Also I started a discussion on all that matter in #1241, don't hesitate to chime in.